src/Entities/Player.cpp

```diff
@@ -1604,12 +1604,18 @@ unsigned int cPlayer::AwardAchievement(const eStatistic a_Ach)
 
 void cPlayer::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
 {
+	// Clamp the positions to exactly representable single-precision floating point values
+	// This is necessary to avoid rounding errors in the noise generator and overflows in the chunk loader
+	const double ClampedPosX = Clamp(a_PosX, -167770000.0, 167770000.0);
+	const double ClampedPosY = Clamp(a_PosY, -167770000.0, 167770000.0);
+	const double ClampedPosZ = Clamp(a_PosZ, -167770000.0, 167770000.0);
```

Comment:
I want at least a constant for these numbers.
If you could represent them with [std::numeric_limits](http://en.cppreference.com/w/cpp/types/numeric_limits), it would be even better :smile: .

P.S.: Maybe also just update the `a_Pos(X|Y|Z)` variables to prevent anyone mixing the clamped and non clamped versions.